### PR TITLE
test: add testcontainers

### PR DESCRIPTION
**Pull request by luky116**
### Purpose or design rationale of this PR

use testcontainer instead of start docker engine by cmd

https://github.com/scroll-tech/scroll/issues/436

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [ ] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [x] test: Adding missing tests or correcting existing tests

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [x] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [ ] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

File: database/migrate/migrate_test.go [hunk 1]
```diff
@@ -7,44 +7,51 @@ import (
 	_ "github.com/lib/pq"
 	"github.com/stretchr/testify/assert"
 
-	"scroll-tech/common/docker"
-
 	"scroll-tech/database"
+	db "scroll-tech/database"
```
Comment (georgehao) [inline database/migrate/migrate_test.go]: why add a duplicated one

File: database/migrate/migrate_test.go [hunk 2]
```diff
@@ -7,44 +7,51 @@ import (
...
+
+	"scroll-tech/common/testcontainers"
+	tc "scroll-tech/common/testcontainers"
 )
 
 var (
-	base *docker.App
-	pgDB *sqlx.DB
+	testApps *testcontainers.TestcontainerApps
+	pgDB     *sqlx.DB
```
Comment (georgehao) [inline database/migrate/migrate_test.go]: rm sqlx

File: rollup/internal/controller/relayer/relayer_test.go [hunk 3]
```diff
@@ -8,25 +8,26 @@ import (
...
 	"scroll-tech/common/database"
-	"scroll-tech/common/docker"
 	dockercompose "scroll-tech/common/docker-compose/l1"
+	"scroll-tech/common/testcontainers"
+	tc "scroll-tech/common/testcontainers"
 	"scroll-tech/common/types/encoding"
 	"scroll-tech/common/types/encoding/codecv0"
 
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/ethclient"
```
Comment (georgehao) [inline rollup/internal/controller/relayer/relayer_test.go]: format import

File: database/migrate/migrate_test.go (hunk 1 above)
Comment (luky116) [reply database/migrate/migrate_test.go]: done

File: rollup/internal/controller/relayer/relayer_test.go (hunk 3 above)
Comment (luky116) [reply rollup/internal/controller/relayer/relayer_test.go]: done

File: database/migrate/migrate_test.go (hunk 2 above)
Comment (luky116) [reply database/migrate/migrate_test.go]: done

File: rollup/internal/controller/relayer/relayer_test.go [hunk 4]
```diff
@@ -14,8 +14,9 @@ import (
 	"github.com/stretchr/testify/assert"
 
 	"scroll-tech/common/database"
-	"scroll-tech/common/docker"
 	dockercompose "scroll-tech/common/docker-compose/l1"
+	"scroll-tech/common/testcontainers"
+	tc "scroll-tech/common/testcontainers"
```
Comment (georgehao) [inline rollup/internal/controller/relayer/relayer_test.go]: duplicated

File: rollup/internal/controller/watcher/watcher_test.go [hunk 5]
```diff
@@ -11,9 +11,9 @@ import (
 	"gorm.io/gorm"
 
 	"scroll-tech/common/database"
-	"scroll-tech/common/docker"
+	"scroll-tech/common/testcontainers"
+	tc "scroll-tech/common/testcontainers"
```
Comment (georgehao) [inline rollup/internal/controller/watcher/watcher_test.go]: ditto

File: rollup/internal/controller/relayer/relayer_test.go (hunk 4 above)
Comment (luky116) [reply rollup/internal/controller/relayer/relayer_test.go]: done

File: rollup/internal/controller/watcher/watcher_test.go (hunk 5 above)
Comment (luky116) [reply rollup/internal/controller/watcher/watcher_test.go]: done

**Review by georgehao — APPROVED** [review]

**Review by colinlyguo — APPROVED** [review]

**Pull request merged**